android/MobiledgeXSDKDemo/computervision/src/main/java/com/mobiledgex/computervision/ImageSender.java

```diff
@@ -56,6 +56,7 @@
 public class ImageSender {
     private static final String TAG = "ImageSender";
     public static final int TRAINING_COUNT_TARGET = 10;
+    private static final double RECOGNITION_CONFIDENCE_THRESHOLD = 105;
```

Comment:
Why 105?

Reply:
Is it not on a 0-1 scale?

Reply:
This has always been confusing. The "confidence" value is actually a "distance" from the trained image that the submitted image most closely matches. So "0" means a perfect match, but there is actually no upper limit. 

105 was arrived at through trial and error. Note that this threshold was previously on the server side, and if reached, the subject name returned was "Unknown". I decided to always return the matched subject name even if the confidence value isn't good, and let the client decide whether to show the subject name, "Unknown", or as we do now add "[Doubtful]" to the returned subject.

The highest value I've seen is over 200. Again, there's no actual upper limit, so it's difficult to come up with a "x% confidence" value.

Reply:
I guess the threshold is for some kind of curve where only the bottom part matters?